topics/admin/tutorials/upgrading/slides.html

```diff
@@ -0,0 +1,170 @@
+topic_name: admin
+tutorial_name: upgrading
+logo: "GTN"
+---
+
+# Release Cycle
+
+* Galaxy aims to release each 4 months
+* Releases are tagged with year and month, for example 17.09, 18.01
+* Per the [security policy](https://github.com/galaxyproject/galaxy/blob/dev/SECURITY_POLICY.md), releases within the past 12 months are supported
+
+---
+# Releases
+
+* Every release has [release notes](https://docs.galaxyproject.org/en/master/releases)
+  * We put substantial effort in making them as useful as we can
+  * Highlights, security announcements, deprecation notices, enhancements grouped by impact, etc...
+* Every release has its own branch in the `galaxy` GitHub repository
+  * Named as `release_YY_MM`
+  * Kept up to date (especially for recent releases)
+
+---
+# Mantain local Galaxy modifications
+
+Two options:
+1. `git stash && git ... && git stash pop`
+2. Commit your changes to a local branch, merge/rebase upstream Galaxy
+
+The latter is probably better in case of large changes
+
+---
+# Keeping a release up to date
+
+To keep your `release_*` branch up to date you can:
+
+```console
+$ git stash  # optional
+$ git pull --ff-only
+$ git stash pop  # optional
+```
+
+and restart Galaxy
+
+Works well if no local commits exist
+
+---
+# Major release upgrade
+
+When a new release is out:
+* Plan a service downtime for the upgrade and inform your users
+* Configure your reverse proxy to serve a custom error page, e.g. for nginx inside the `server` section add:
+  ```ini
+  error_page 502 /static/custom_502.html;
+  ```
+* When it's time, stop the Galaxy server processes (not the database server!)
+
+---
+# Housekeeping
+
+Not usually necessary, but might help:
+
+```console
+$ find . -name '*.pyc' -delete
+$ rm -rf database/compiled_templates/*
+```
+
+---
+# Upgrading the galaxy repo
+
+```console
+$ git fetch
+$ git stash  # optional
+$ git checkout release_YY.MM
+$ git pull --ff-only
+$ git stash pop  # optional
+```
+
+---
+# Diff samples
+
+```console
+$ diff -u galaxy.ini galaxy.ini.sample
+$ diff -u datatypes_conf.xml datatypes_xml.conf.sample
+```
+
+Merge changes as desired/necessary.
+
+???
+Alternatively:
+
+```console
+$ git diff release_17.09..release_18.01 -- config/galaxy.ini.sample
+```
+
+---
+# Upgrade virtualenv
+
+```console
+$ export GALAXY_CONFIG_FILE=/srv/galaxy/config/galaxy.ini
+$ export GALAXY_VIRTUAL_ENV=/srv/galaxy/venv
+$ . $GALAXY_VIRTUAL_ENV/bin/activate
+$ pip install --upgrade pip setuptools
+$ ./scripts/common_startup.sh
+$ deactivate
+```
+
+---
+# make client on `dev`
+
+Since January 2018, the `dev` Galaxy branch does not contain updated client
+build artifacts (e.g. JavaScript bundles in `static/`).
+
+1. install `yarn`
+2. run `make client`
+
+---
+# Tool migrations?
+
+Galaxy source tools -> Tool Shed
+
+We haven't done these in a long time, but may do more
+
+Galaxy will notify you on first startup after upgrade including migration
+
+---
+# Database migrations
+
+1. **Backup** your database
```

Comment:
Imho this should be in preparation, see above.

Reply:
I think the idea is that a backup is needed only if there are migrations, ping @martenson.

Reply:
@nsoranzo that was the original point, yes

Reply:
I do think a backup is always needed, can never hurt and should be done before anything else. But this is not blocking this PR :)

Reply:
I agree, but this is specifically database migration slide. Feel free to add it above of course.